app/Http/Controllers/Api/Web/AlbumsController.php

```diff
@@ -83,6 +86,53 @@ public function getShow($id)
         ], 200);
     }
 
+    public function getCachedAlbum($id, $format)
+    {
+        // Validation
+        try {
+            $album = Album::findOrFail($id);
+        } catch (ModelNotFoundException $e) {
+            return $this->notFound('Album not found!');
+        }
+
+        if (!array_key_exists($format, Track::$CacheableFormats)) {
+            return $this->notFound('Format not found!');
+        }
+
+        $tracks = $album->tracks;
+
+        $trackCount = 0;
+        $cachedCount = 0;
+
+        foreach($tracks as $track) {
+            if ($track->is_downloadable == false) {
+                continue;
+            } else {
+                $trackCount++;
+            }
+
+            try {
+                $trackFile = $track->trackFiles()->where('format', $format)->firstOrFail();
```

Comment:
Another opportunity to use eager loading - n+1 queries are being generated here.